mailman/patchset.py

```diff
@@ -56,6 +57,24 @@ def am_cover_letter(keep_empty=True):
     for i, patch in enumerate(patches):
         patch_abspath = str(maildir / patch.msg_id)
 
+        with open(patch_abspath, 'r') as patch_file:
+            patch_content = patch_file.read()
+        match = re.search(r'^\s+Signed-off-by:\s+.+\s+<(.+)@.+>$', patch_content, re.MULTILINE)
+        if match:
+            found_author = match.group(1)
+        else:
+            return f'illegal patch {i+1}: missing Signed-off-by line!'
+
+        changelines = list(filter(lambda line: line.startswith('--- ') or line.startswith('+++ '), patch_content.split('\n')))
+        for change in changelines:
+            file = change.split(' ')[1].strip()
+            if file == '/dev/null':
+                continue
+            first_dir = file.split('/')[1]
+            if first_dir != found_author:
```

Comment:
I guess you want the author so you can do this... but I feel like you should either have the gradable so you can just check `.user` or look at the same source of truth as mailman and examine the transaction snippet that lists the username.